### PR TITLE
Impact Study (take 2)

### PR DESCRIPTION
**Pull request by mulark**
Well after blundering it, recreated

### CODE REVIEW

**Comment by mulark** [comment]
dmidecode considers string index 0 to be "Not Specified", and smbios-lib considers it an empty string.

File: src/default_out.rs [hunk 1]
```diff
@@ -85,24 +85,16 @@ pub fn dump_undefined_struct(
...
-                            }
-                            ExtendedRomSize::Gigabytes(size) => {
-                                println!("{} {}", size, GB);
-                            }
-                            ExtendedRomSize::Undefined(size) => {
-                                println!("{} ({})", OUT_OF_SPEC, size);
-                            }
-                        }
-                    }
+            let rom_size = data.extended_rom_size().or(data.rom_size());
```
Comment (jrgerber) [inline src/default_out.rs:88]: I like the idea of using the .or() on these extended cases.  We can borrow this pattern in the dmi_cache_size() code refactoring I did and reduce some of the match arms.

File: src/dmifn.rs [hunk 2]
```diff
@@ -2101,6 +2103,12 @@ pub fn dmi_on_board_devices_type(device_type: &OnBoardDeviceType) -> String {
         TypeOfDevice::PataController => "PATA Controller",
         TypeOfDevice::SataController => "SATA Controller",
         TypeOfDevice::SasController => "SAS Controller",
+        TypeOfDevice::WirelessLan => "Wireless LAN",
+        TypeOfDevice::Bluetooth => "Bluetooth",
+        TypeOfDevice::Wwan => "WWAN",
+        TypeOfDevice::Emmc => "eMMC (embedded Milti-Media Controller)",
```
Comment (jrgerber) [inline src/dmifn.rs:2109]: Milti?  spelling?

**Comment by jrgerber** [comment]
> dmidecode considers string index 0 to be "Not Specified", and smbios-lib considers it an empty string.

Ah, interesting.  I haven't looked at the original C code on this.  So, parity would be an output of "Not Specified" for all strings?

This sounds like there is a need for a single function that prints an SMBiosString and takes care of both calling the .to_utf_lossy() and printing "Not Specified" for the None case.

**Comment by mulark** [comment]
> > dmidecode considers string index 0 to be "Not Specified", and smbios-lib considers it an empty string.
> 
> Ah, interesting. I haven't looked at the original C code on this. So, parity would be an output of "Not Specified" for all strings?
> 
> This sounds like there is a need for a single function that prints an SMBiosString and takes care of both calling the .to_utf_lossy() and printing "Not Specified" for the "" case.

Yep, and it also considers OOB string accesses to be `<BAD INDEX>` I'll have a commit shortly that matches the behavior on strings.

**Pull request merged**